diff --git a/prisma-client/generator-build/index.js b/prisma-client/generator-build/index.js
index d443b0846a56095f72e0a1d284dc807a918db7fd..2e7ec96f05e6d830d70c6db43742b697c8758d54 100644
--- a/prisma-client/generator-build/index.js
+++ b/prisma-client/generator-build/index.js
@@ -7243,10 +7243,17 @@ if (!fs.existsSync(path.join(__dirname, 'schema.prisma'))) {
   
   const alternativePath = alternativePaths.find((altPath) => {
     return fs.existsSync(path.join(process.cwd(), altPath, 'schema.prisma'))
-  }) ?? alternativePaths[0]
+  }); 
 
-  config.dirname = path.join(process.cwd(), alternativePath)
-  config.isBundled = true
+  if(alternativePath) {
+    config.dirname = path.join(process.cwd(), alternativePath)
+  } else if(config?.generator?.output?.value) {
+    config.dirname = config.generator.output.value;
+  } else {
+    config.dirname = path.join(process.cwd(), alternativePaths[0]);
+  }
+
+  config.isBundled = true  
 }`;
 }
 function buildDirnameDefault() {
